File: src/cosmic_pi_network/cvlib/object_detection.py
import cv2
import numpy as np

class ObjectDetection:
    def __init__(self, yolo_v4: YOLOv4):
        self.yolo_v4 = yolo_v4

    def detect_objects(self, image: np.ndarray) -> list:
        return self.yolo_v4.detect(image)

File: src/cosmic_pi_network/cvlib/cpp/ObjectDetection.cpp
#include "ObjectDetection.h"

ObjectDetection::ObjectDetection(YOLOv4* yolo_v4) : yolo_v4_(yolo_v4) {}

std::vector<cv::Mat> ObjectDetection::detect_objects(const cv::Mat& image) {
    return yolo_v4_->detect(image);
}
